File: backend/src/main/java/pt/ist/socialsoftware/mono2micro/strategy/domain/StrategyFactory.java
package pt.ist.socialsoftware.mono2micro.strategy.domain;

import static pt.ist.socialsoftware.mono2micro.strategy.domain.AccessesSciPyStrategy.ACCESSES_SCIPY;
import static pt.ist.socialsoftware.mono2micro.strategy.domain.RepositorySciPyStrategy.REPOSITORY_SCIPY;

public class StrategyFactory {
    private static StrategyFactory factory = null;

    public static StrategyFactory getFactory() {
        if (factory == null)
            factory = new StrategyFactory();
        return factory;
    }

    public Strategy getStrategy(String strategyType) {
        switch (strategyType) {
            case ACCESSES_SCIPY:
                return new AccessesSciPyStrategy();
            case REPOSITORY_SCIPY:
                return new RepositorySciPyStrategy();
            default:
                throw new RuntimeException("The type \"" + strategyType + "\" is not a valid strategy type.");
        }
    }
}

File: frontend/src/models/strategy/StrategyFactory.ts
import Strategy, { StrategyType } from "../strategy/Strategy";
import AccessesSciPyStrategy from "./AccessesSciPyStrategy";
import RepositorySciPyStrategy from "./RepositorySciPyStrategy";
import AccAndRepoSciPyStrategy from "./AccAndRepoSciPyStrategy";

export abstract class StrategyFactory {
    static getStrategy(strategy: any) : Strategy {
        switch (strategy.type) {
            case StrategyType.ACCESSES_SCIPY:
                return new AccessesSciPyStrategy(strategy);
            case StrategyType.REPOSITORY_SCIPY:
                return new RepositorySciPyStrategy(strategy);
            case StrategyType.ACC_AND_REPO_SCIPY:
                return new AccAndRepoSciPyStrategy(strategy);
            default:
                throw new Error('Type ' + strategy.type + ' unknown.');
        }
    }
}
